pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -176,8 +244,11 @@ func (p *DNSProxy) checkRestored(endpointID uint64, destPort uint16, destIP stri
 	}
 
 	for i := range ipRules {
-		if _, exists := ipRules[i].IPs[destIP]; (exists || ipRules[i].IPs == nil) && ipRules[i].Re.MatchString(name) {
-			return true
+		if _, exists := ipRules[i].IPs[destIP]; exists || ipRules[i].IPs == nil {
+			_, ok := ipRules[i].FQDNs[name]
+			if ok || ipRules[i].Re.MatchString(name) {
```

Comment:
Hey. Yee, if we want it to support that, we would have to nil check here. We can do that, but ref. the other comment, then we probably need to backport some changes so that downgrading doesn't cause nil pointer dereferences.

Not sure about what versioning guarantees are given in cilium. Eg. if we can ship this as version `v1.X.0`, and it supports downgrade to `v1.(X-1).9` but not eg. `v1.(X-1).8`.


But happy to implement that if thats something you want. 😄

Reply:
Yes, we can only add support for downgrade for one previous version. Is that what you were asking about?

Reply:
Nice! Did some testing now, and it turns out it should be fairly ok to implement support for it. I'll add support for `null` values for the FQDN map as well to keep it consistent. 😄